server/src-lib/Hasura/GraphQL/Transport/WebSocket.hs

```diff
@@ -404,22 +411,26 @@ onStart env serverEnv wsConn (StartMsg opId q) = catchAndIgnore $ do
       -> Maybe EQ.GeneratedSqlMap
       -> RequestId
       -> ExceptT QErr (ExceptT () m) EncJSON
-      -> ExceptT () m ()
+      -> ExceptT () m EncJSON
     execQueryOrMut timerTot telemQueryType telemCacheHit genSql requestId action = do
       let telemLocality = Telem.Local
       logOpEv ODStarted (Just requestId)
       -- log the generated SQL and the graphql query
       logQueryLog logger q genSql requestId
-      withElapsedTime (runExceptT action) >>= \case
-        (_,      Left err) -> postExecErr requestId err
+      result <- withElapsedTime (runExceptT action) >>= \case
+        (_,      Left err) -> do
+          postExecErr requestId err
+          throwError ()
```

Comment:
The `completed` message has to be sent even when there is an error. So `throwError` shouldn't be called, Instead the type of this function can be `ExceptT () m (Maybe EncJSON)`? i.e, only the successful response is cached.

Reply:
I'm not very sure why this function has the signature `ExceptT () ..`.

Reply:
I fixed it using your suggestion; not sure why that function has ExceptT either, I'll have a look at whether it can be simplified.